airflow/providers/sqlite/hooks/sqlite.py

```diff
@@ -33,13 +34,21 @@ class SqliteHook(DbApiHook):
 
     def get_conn(self) -> sqlite3.dbapi2.Connection:
         """Returns a sqlite connection object"""
-        conn_id = getattr(self, self.conn_name_attr)
-        airflow_conn = self.get_connection(conn_id)
-        conn = sqlite3.connect(airflow_conn.host)
+        # The sqlite3 connection does not use the sqlite scheme and path should not end with a slash.
+        # See https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#uri-connections for details.
+        sqlalchemy_uri = self.get_uri()
+        sqlite_uri = sqlalchemy_uri.replace("sqlite:///", "file:").replace("/?", "?")
```

Comment:
The trailing slash in the path is a side effect of using `airflow_conn.get_uri()` 😞 I can also move the replace on this function call instead.

Reply:
I am not quite sure, because the URI format of an airflow connection is different from the one [SQLite uses](https://www.sqlite.org/uri.html).

Reply:
SQLite `file:` URIs can't end with a slash because they are supposed to be files - not directories, but an airflow connection path can be anything.